QuoridorEngine/src/core/QuoridorGame.cs

```diff
@@ -118,6 +155,66 @@ public bool HasWall(int row, int column, Orientation orientation)
             throw new ArgumentException("Unknown orientation type");
         }
 
+        /// <summary>
+        /// Check whether provided wall placement move is valid
+        /// </summary>
+        /// <param name="move">Wall placement move</param>
+        /// <returns>True if valid</returns>
+        public bool CanPlaceWall(QuoridorMove move)
```

Comment:
Don't think this needs to be public. If any client of this class wants to check if a move is legal they can just send the move in for execution and catch the InvalidMoveException.

Make it private and move it down